packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -390,28 +394,25 @@ ValueResult<uint256_t> ArbCore::unexpiredMessageCount() {
 // reorgToMessageCountOrBefore resets the checkpoint and database entries
 // such that machine state is at or before the requested message. cleaning
 // up old references as needed.
-// If use_latest is true, message_sequence_number is ignored and the latest
-// checkpoint is used.
+// If initial_start is true the various caching data structures are seeded.
 rocksdb::Status ArbCore::reorgToMessageCountOrBefore(
-    const uint256_t& message_count,
+    const std::function<bool(const MachineOutput&)>& check_output,
     bool initial_start,
     ValueCache& cache) {
     std::variant<std::unique_ptr<MachineThread>, rocksdb::Status> setup =
         rocksdb::Status::OK();
 
     if (initial_start) {
-        std::cerr << "Reloading cache starting with message " << message_count
-                  << std::endl;
+        std::cerr << "Reloading cache" << std::endl;
```

Comment:
Having the message count here was nice, so maybe it'd be good to have `reorgToMessageCountOrBefore` as a wrapper around `reorgCheckpoints` and include the prints as they were

Reply:
:+1: